contracts/abstract/JBNFTRewardDataSource.sol

```diff
@@ -179,62 +238,22 @@ abstract contract JBNFTRewardDataSource is
 
   /**
     @notice 
-    Part of IJBFundingCycleDataSource, this function gets called when the project receives a payment. It will mint an NFT to the contributor (_data.beneficiary) if conditions are met.
+    Part of IJBPayDelegate, this function gets called when the project receives a payment. It will mint an NFT to the contributor (_data.beneficiary) if conditions are met.
 
     @dev 
-    This function will revert if the contract calling it is not the store of one of the project's terminals. 
+    This function will revert if the contract calling is not one of the project's terminals. 
 
     @param _data The Juicebox standard project contribution data.
-
-    @return weight The weight that tokens should get minted in accordance to 
-    @return memo The memo that should be forwarded to the event.
-    @return delegate A delegate to call once the payment has taken place.
   */
-  function payParams(JBPayParamsData calldata _data)
-    external
-    override
-    returns (
-      uint256 weight,
-      string memory memo,
-      IJBPayDelegate delegate
-    )
-  {
-    // Make sure the caller is expected, and the call is being made on behalf of an interaction with the correct project.
+  function didPay(JBDidPayData calldata _data) external override {
+    // Make sure the caller is a terminal of the project, and the call is being made on behalf of an interaction with the correct project.
     if (
-      !directory.isTerminalOf(projectId, _data.terminal) ||
-      msg.sender != _expectedCaller ||
+      !directory.isTerminalOf(projectId, IJBPaymentTerminal(msg.sender)) ||
```

Comment:
msg.sender is the terminal store!

Reply:
not for the delegate! The delegate gets called from the terminal proper.

Reply:
but this is payParams, isn't it the dataSource?

Reply:
oh no, mb, this became didPay, nvm